src/components/App/index.tsx

```diff
@@ -1,20 +1,26 @@
-import React, { useState } from 'react';
+import React, { useEffect } from 'react';
 import i18next from 'i18next';
 
+import { Context, reducer, INITIAL_STATE, useSelector } from 'contexts';
 import Routes from 'routes';
 import I18n from 'components/I18n';
+import withProvider from 'components/ProviderWrapper';
 
 import styles from './styles.module.scss';
 
 function App() {
-  const [language, setLanguage] = useState(i18next.language);
+  const language = useSelector(state => state.language);
+
+  useEffect(() => {
+    i18next.changeLanguage(language);
+  }, [language]);
 
   return (
     <div className={`${styles.appContainer} row center middle`}>
-      <I18n language={language} setLanguage={setLanguage} />
+      <I18n />
       <Routes />
     </div>
   );
 }
 
-export default App;
+export default withProvider({ context: Context, reducer, initialState: INITIAL_STATE })(App);
```

Comment:
Since this is not a regular component but rather the App component I would prefer you to use the Context.Provider component instead of using HOC